apps/passport-client/components/screens/LoginScreens/LoginScreen.tsx

```diff
@@ -164,23 +181,28 @@ export function LoginScreen(): JSX.Element {
    */
   const requestStorageAndLogIn = useCallback(async () => {
     try {
+      setStorageAccessStatus(StorageAccessStatus.Requesting);
       // @ts-expect-error Chrome-only API
       const handle: { localStorage: Storage } =
         // @ts-expect-error Chrome-only API
         await document.requestStorageAccess({ localStorage: true });
 
+      setStorageAccessStatus(StorageAccessStatus.Granted);
       // Access granted, try reading the local storage
       const encryptionKey = handle.localStorage.getItem("encryption_key");
       if (encryptionKey) {
         await tryToLogin(encryptionKey);
+      } else {
+        setStorageAccessStatus(StorageAccessStatus.NoLocalStorage);
       }
     } catch (_e) {
-      // If the user rejected the storage access request, catch the exception
-      // but otherwise do nothing. The finally block will return the user to
-      // the regular login flow.
+      // If the user rejected the storage access request, set an error message.
+      // The finally block will return the user to the regular login flow.
+      setError(
+        "Unable to log in automatically, please enter your email to log in"
+      );
     } finally {
-      setTryStorageAccess(false);
-      setTryStorageLogin(false);
+      setStorageAccessStatus(StorageAccessStatus.Denied);
```

Comment:
This is going to override values set in various codepaths above (including Success or NoLocalStorage).
I think this line belongs in the catch block, not the finally block.
Or maybe you want it to be `if (state===Requesting) { state = Denied; }` as a fallback, but at the moment I don't think there's a non-exception case which would trigger that.

Reply:
Yes, this belongs in the catch block - I've moved it. Thanks!